### PR TITLE
Virtual keyboard

### PR DESCRIPTION
**Pull request by daklin2**
**1.** [Task](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/codejam-virtual-keyboard.md) description

**2.** Upload [site]( https://daklin2.github.io/virtual-keyboard/)

**3.** Screenshot
Screen Shot 2020-04-02 at 8 27 56 AM

**4.**
Deadline | Branch name
-- | --
07.04.2020 23:59 | gh-pages

Total score: 110/110
Minimum:

  1. creating DOM-elements in empty index.html   +20/20
  2. pressing the button on real keyboard highlighted button in virtual keyboard   +10/10
Standard

 1. language changing and saving selected language   +15/15
 2. clicking on virtual keyboard enter symbols in input   +15/15
Level up

 1. animated pushing on buttons   +15/15

Technical requirements:

 1. using ES6 issues (classes, destructurization)   +15/15
 2. using eslint   +10/10
 3. requirements for repository, commits and PR are implemented   +10/10
Exactions

1. errors of warnings of eslint-config-airbnb-base   -15/0

shift + alt - another language

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        row.forEach((keyValue) => {
+          if (keyValue[1] === targetBtnName && (specialBtn === undefined || specialBtn === 'space')) {
+            const [, , ruLow, ruUp, enLow, enUp] = keyValue;
+            switch (this.lang) {
+              case 'en':
+                if (this.isShiftPress || this.isCapsLockPress) this.symbol = enUp;
+                else this.symbol = enLow;
+                break;
+              case 'ru':
+                if (this.isShiftPress || this.isCapsLockPress) this.symbol = ruUp;
```
Comment (san4ezgig) [inline src/main.js]: Use brackets for each if statement, it improves readability

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        }
+      }
+
+      if (!this.isShiftPress && specialBtn === 'capslock') {
+        this.checkCaps();
+      }
+
+      if (specialBtn === 'arrow') {
+        const pos = this.textArea.selectionStart;
+        if (targetBtnName === 'ArrowUp') this.setCaretPosition(pos - this.textArea.value.length);
```
Comment (san4ezgig) [inline src/main.js]: switch case

**Review by san4ezgig — COMMENTED** [review]
110

File: src/main.js (hunk 2 above)
Comment (daklin2) [reply src/main.js]: fixed =)

File: src/main.js (hunk 1 above)
Comment (daklin2) [reply src/main.js]: now exactly got what that means and fixed =)

**Pull request merged**